distributed/deploy/adaptive_core.py

```diff
@@ -178,6 +178,12 @@ def stop(self):
         self._adapting = True
 
         try:
+
+            if len(self.plan) != len(self.requested):
+                # Ensure that the number of planned and requested workers
+                # stay in sync.
+                await self.cluster
```

Comment:
Do we require that `self.cluster` be awaitable for every backend?

Reply:
That's a good point!

I've moved this logic into the main `Adaptive` class where there definitely should be a cluster provided, but I'm not sure what the best way to check if the cluster object is awaitable. Really this should only be needed if the cluster is a `SpecCluster`, but I think checking for that here will introduce a circular dependency?

Reply:
I *believe* that this is a safe assumption.  Certainly SpecCluster does this, and I think that dask-labextension expects it, so probably @jcrist has enforced this on dask-yarn and dask-gateway.

Reply:
Dask-gateway is awaitable, but doesn't support adaptive yet. dask-yarn is currently not async at all, but should be (TODO). I consider the async interface that `distributed.deploy.Cluster` defines to be the proper spec, so anything that supports should be fine.